From 4cbb588b229eeaac69b7b024c9718135a616c9e1 Mon Sep 17 00:00:00 2001
From: KoffeinFlummi <koffeinflummi@protonmail.com>
Date: Sat, 4 May 2019 19:47:24 +0200
Subject: [PATCH 2/3] Make changes for OTA updates

---
 phhgsi_arm64_ab/BoardConfig.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/phhgsi_arm64_ab/BoardConfig.mk b/phhgsi_arm64_ab/BoardConfig.mk
index 2b59f0f..ef807de 100644
--- a/phhgsi_arm64_ab/BoardConfig.mk
+++ b/phhgsi_arm64_ab/BoardConfig.mk
@@ -1,6 +1,10 @@
 include build/make/target/board/generic_arm64_ab/BoardConfig.mk
 include device/phh/treble/board-base.mk
 
+AB_OTA_UPDATER := true
+AB_OTA_PARTITIONS := system
+PRODUCT_PACKAGES += update_engine update_verifier
+
 ifeq ($(BOARD_SYSTEMIMAGE_PARTITION_RESERVED_SIZE),)
 BOARD_SYSTEMIMAGE_PARTITION_SIZE := 2147483648
 endif
-- 
2.21.0

