From 3643085a4ba393f224677f65027d19f8aa959395 Mon Sep 17 00:00:00 2001
From: KoffeinFlummi <koffeinflummi@gmail.com>
Date: Sat, 28 Jul 2018 13:04:02 +0200
Subject: [PATCH 1/2] Adjust core Makefile

---
 target/product/core.mk | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/target/product/core.mk b/target/product/core.mk
index a2b0f1c55..657f62d50 100644
--- a/target/product/core.mk
+++ b/target/product/core.mk
@@ -24,7 +24,7 @@ PRODUCT_PACKAGES += \
     BasicDreams \
     BlockedNumberProvider \
     BookmarkProvider \
-    Browser2 \
+    chromium \
     BuiltInPrintService \
     Calendar \
     CalendarProvider \
@@ -42,7 +42,7 @@ PRODUCT_PACKAGES += \
     KeyChain \
     Keyguard \
     LatinIME \
-    Launcher2 \
+    Launcher3 \
     ManagedProvisioning \
     MtpDocumentsProvider \
     PicoTts \
@@ -51,7 +51,6 @@ PRODUCT_PACKAGES += \
     PrintSpooler \
     PrintRecommendationService \
     ProxyHandler \
-    QuickSearchBox \
     Settings \
     SharedStorageBackup \
     StorageManager \
@@ -59,7 +58,9 @@ PRODUCT_PACKAGES += \
     TeleService \
     VpnDialogs \
     vr \
-    MmsService
+    MmsService \
+    SysuiDarkThemeOverlay \
+    PdfViewer
 
 # The set of packages whose code can be loaded by the system server.
 PRODUCT_SYSTEM_SERVER_APPS += \
-- 
2.18.0

