From fc8aaf1781d0caa26f07260ce3062e114ede8729 Mon Sep 17 00:00:00 2001
From: KoffeinFlummi <koffeinflummi@gmail.com>
Date: Wed, 29 Aug 2018 00:32:22 +0200
Subject: [PATCH 2/2] Update microg GmsCore

---
 update.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/update.sh b/update.sh
index 2b39e5a..434c607 100644
--- a/update.sh
+++ b/update.sh
@@ -64,7 +64,7 @@ downloadFromFdroid com.github.yeriomin.yalpstore "" "true"
 downloadFromFdroid org.fitchfamily.android.dejavu
 downloadFromFdroid org.fdroid.fdroid
 
-wget https://microg.org/fdroid/repo/com.google.android.gms-12221403.apk -O bin/com.google.android.gms.apk
+wget https://microg.org/fdroid/repo/com.google.android.gms-12879000.apk -O bin/com.google.android.gms.apk
 addSystemCopy com.google.android.gms.apk com.google.android.gms
 
 wget https://microg.org/fdroid/repo/com.google.android.gsf-8.apk -O bin/com.google.android.gsf.apk
-- 
2.18.0

