From dafb684f1c92644dff8e52df553bc971205e9b2a Mon Sep 17 00:00:00 2001
From: KoffeinFlummi <koffeinflummi@gmail.com>
Date: Fri, 28 Sep 2018 15:49:21 +0200
Subject: [PATCH 3/3] Update microg GmsCore to v0.2.6

---
 update.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/update.sh b/update.sh
index 434c607..fb7e726 100644
--- a/update.sh
+++ b/update.sh
@@ -64,7 +64,7 @@ downloadFromFdroid com.github.yeriomin.yalpstore "" "true"
 downloadFromFdroid org.fitchfamily.android.dejavu
 downloadFromFdroid org.fdroid.fdroid
 
-wget https://microg.org/fdroid/repo/com.google.android.gms-12879000.apk -O bin/com.google.android.gms.apk
+wget https://microg.org/fdroid/repo/com.google.android.gms-13280012.apk -O bin/com.google.android.gms.apk
 addSystemCopy com.google.android.gms.apk com.google.android.gms
 
 wget https://microg.org/fdroid/repo/com.google.android.gsf-8.apk -O bin/com.google.android.gsf.apk
-- 
2.19.0

