From a312fb80b9b10049c75a8c251a785fd69031c74d Mon Sep 17 00:00:00 2001
From: KoffeinFlummi <koffeinflummi@gmail.com>
Date: Sat, 28 Jul 2018 13:22:52 +0200
Subject: [PATCH] Disable permissive domain check for user builds

---
 Android.mk | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/Android.mk b/Android.mk
index f0c6a6464..24456afb4 100644
--- a/Android.mk
+++ b/Android.mk
@@ -702,13 +702,13 @@ $(built_sepolicy_neverallows)
 	@mkdir -p $(dir $@)
 	$(hide) $< -m -M true -G -c $(POLICYVERS) $(PRIVATE_NEVERALLOW_ARG) $(PRIVATE_CIL_FILES) -o $@.tmp -f /dev/null
 	$(hide) $(HOST_OUT_EXECUTABLES)/sepolicy-analyze $@.tmp permissive > $@.permissivedomains
-	$(hide) if [ "$(TARGET_BUILD_VARIANT)" = "user" -a -s $@.permissivedomains ]; then \
-		echo "==========" 1>&2; \
-		echo "ERROR: permissive domains not allowed in user builds" 1>&2; \
-		echo "List of invalid domains:" 1>&2; \
-		cat $@.permissivedomains 1>&2; \
-		exit 1; \
-		fi
+	#$(hide) if [ "$(TARGET_BUILD_VARIANT)" = "user" -a -s $@.permissivedomains ]; then \
+	#	echo "==========" 1>&2; \
+	#	echo "ERROR: permissive domains not allowed in user builds" 1>&2; \
+	#	echo "List of invalid domains:" 1>&2; \
+	#	cat $@.permissivedomains 1>&2; \
+	#	exit 1; \
+	#	fi
 	$(hide) mv $@.tmp $@
 
 built_sepolicy := $(LOCAL_BUILT_MODULE)
-- 
2.18.0

